File: java-suite/src/main/java/de/fraunhofer/iem/classesAndObjects/classes/inheritance/baseDerivedWithMultipleConstructor/Cat.java
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.baseDerivedWithMultipleConstructor;

/**
 * Inherit the Animal class and can override makeNoise method
 * <p>
 * Note:
 * 1. Two constructor calls different super class constructor
 *
 * @author Ranjith Krishnamurthy
 */
public class Cat extends Animal {
    public Cat(String DNA) {
        super(DNA);
    }

    public Cat() {
        super();
    }

    @Override
    public void makeNoise() {
        System.out.println("Meow");
    }
}

File: kotlin-suite/src/main/kotlin/de/fraunhofer/iem/classesAndObjects/classes/inheritance/baseDerivedWithMultipleConstructor/Cat.kt
package de.fraunhofer.iem.classesAndObjects.classes.inheritance.baseDerivedWithMultipleConstructor

/**
 * Inherit the Animal class and can override makeNoise method
 *
 *
 * Note:
 * 1. Two constructor calls different super class constructor
 *
 * @author Ranjith Krishnamurthy
 */
class Cat : Animal {
    constructor(DNA: String) : super(DNA)

    constructor() : super()

    override fun makeNoise() {
        println("Meow")
    }
}